src/main/java/com/zendesk/maxwell/schema/SchemaCapturer.java

```diff
@@ -261,4 +269,13 @@ private void captureTablesPK(Database db, HashMap<String, Table> tables) throws
 		return result.toArray(new String[0]);
 	}
 
+	@Override
+	public void close() throws SQLException {
+		try (PreparedStatement p1 = tablePreparedStatement;
+			 PreparedStatement p2 = columnPreparedStatement;
+			 PreparedStatement p3 = pkPreparedStatement) {
+			// auto-close shared prepared statements
+		}
+	}
```

Comment:
this is a bit goofy, is there not a `PreparedStatement#close` or something?

Reply:
There is, but I think the alternative when dealing with multiple resources is weirder:
```
...
finally {
  try {
    pkPreparedStatement.close();
  } finally {
    try {
      columnPreparedStatement.close();
    } finally {
      tablePreparedStatement.close();
    }
  }
}
```